File: algorithms/js/1423_maximumPointsYouCanObtainfromCards/maximumPointsYouCanObtainfromCards.js
/**
 * @param {number[]} cardPoints
 * @param {number} k
 * @return {number}
 * 
 * https://leetcode.com/problems/maximum-points-you-can-obtain-from-cards/discuss/1202397/JS-Python-Java-C%2B%2B-or-Easy-Sliding-Window-Solution-w-Explanation
 */

 var maxScore = function(cardPoints, k) {
    let sum = cardPoints.slice(0, k).reduce((a, b) => a + b);
    let res = sum;
    for (let i = k - 1, j = cardPoints.length - 1; i >= 0; i--, j--) {
        sum += cardPoints[j] - cardPoints[i];
        res = Math.max(res, sum);
    }
    return res;
};

File: algorithms/python/maximumPointsYouCanObtainfromCards/maximumPointsYouCanObtainfromCards.py
class Solution:
    def maxScore(self, cardPoints: List[int], k: int) -> int:
        ans = kSum = sum(cardPoints[:k])
        for i in range(k - 1, -1, -1):
            kSum += cardPoints[len(cardPoints) - k + i] - cardPoints[i]
            ans = max(ans, kSum)
        return ans